bazel/python_rules.bzl

```diff
@@ -94,93 +70,155 @@ def _generate_py_impl(context):
             providers = [ProtoInfo],
         ),
-        "plugin": attr.label(
-            executable = True,
-            providers = ["files_to_run"],
-            cfg = "host",
-        ),
-        "flags": attr.string_list(
-            mandatory = False,
-            allow_empty = True,
-        ),
-        "well_known_protos": attr.label(mandatory = False),
         "_protoc": attr.label(
             default = Label("//external:protocol_compiler"),
+            providers = ["files_to_run"],
             executable = True,
             cfg = "host",
         ),
     },
-    output_to_genfiles = True,
     implementation = _generate_py_impl,
 )
 
-def _generate_py(well_known_protos, **kwargs):
-    if well_known_protos:
-        __generate_py(
-            well_known_protos = "@com_google_protobuf//:well_known_protos",
-            **kwargs
-        )
-    else:
-        __generate_py(**kwargs)
-
 def py_proto_library(
         name,
-        deps,
-        well_known_protos = True,
-        proto_only = False,
+        srcs,
         **kwargs):
     """Generate python code for a protobuf.
 
     Args:
       name: The name of the target.
-      deps: A list of dependencies. Must contain a single element.
-      well_known_protos: A bool indicating whether or not to include well-known
-        protos.
-      proto_only: A bool indicating whether to generate vanilla protobuf code
-        or to also generate gRPC code.
+      srcs: A list of proto_library dependencies. Must contain a single element.
     """
-    if len(deps) > 1:
-        fail("The supported length of 'deps' is 1.")
-
     codegen_target = "_{}_codegen".format(name)
-    codegen_grpc_target = "_{}_grpc_codegen".format(name)
+    if len(srcs) > 1:
+        fail("Can only compile a single proto at a time.")
 
-    _generate_py(
+
+    _generate_pb2_src(
         name = codegen_target,
-        deps = deps,
-        well_known_protos = well_known_protos,
+        deps = srcs,
         **kwargs
     )
 
-    if not proto_only:
-        _generate_py(
-            name = codegen_grpc_target,
-            deps = deps,
-            plugin = "//src/compiler:grpc_python_plugin",
-            well_known_protos = well_known_protos,
-            **kwargs
-        )
+    native.py_library(
+        name = name,
+        srcs = [":{}".format(codegen_target)],
+        deps = ["@com_google_protobuf//:protobuf_python"],
+        **kwargs
+    )
 
-        native.py_library(
-            name = name,
-            srcs = [
-                ":{}".format(codegen_grpc_target),
-                ":{}".format(codegen_target),
-            ],
-            deps = [requirement("protobuf")],
-            **kwargs
-        )
-    else:
-        native.py_library(
-            name = name,
-            srcs = [":{}".format(codegen_target), ":{}".format(codegen_target)],
-            deps = [requirement("protobuf")],
-            **kwargs
+def _generate_pb2_grpc_src_impl(context):
```

Comment:
nit: This function seems share a lot of common logic with `_generate_py_impl`. Do you think you we should somehow merge them into one? It might help readability by isolating the difference (between _py_impl & _grpc_src_impl) part as function arguments.